Adding extended setup instructions (#470)
datacarpentry/python-ecology-lesson: setup.md

> ## Installing Python using Anaconda
> [Python][python] is a popular language for scientific computing, and great for

> > general-purpose programming as well. Installing all of its scientific packages

Reviewer: Do you think we should limit the range of "scientific packages" we talk about here to the ones we use in the lesson?

Author: I don't think so. This is just very confusing if you don't really know anything about python. I frequently have this with Ruby which I use very rarely and get a bit overwhelmed with random package names.

Reviewer: I meant something like `Installing all of the scientific packages we use in the lesson individually...`

> individually can be a bit difficult, however, so we recommend the all-in-one
> installer [Anaconda][anaconda].
>
> Regardless of how you choose to install it, please make sure you install Python
> version 3.x (e.g., 3.6 is fine). Also, please set up your Python environment at 
> least a day in advance of the workshop.  If you encounter problems with the 
> installation procedure, ask your workshop organizers via e-mail for assistance so

> > you are ready to go as soon as the workshop begins.

Reviewer: I agree it is useful to remind participants to prepare a bit in advance. But currently it's a bit hidden. Do you have any suggestions how we could make it more visible?

Author: I am not sure where a good place for this would be. I tend to highlight this in an email I send out to participants before a workshop.

Reviewer: We send similar reminders/requests to workshop participants but I'm on the fence whether this should be part of the lesson -- when participants install software seems to be more appropriate for a workshop page where all event-related questions are handled.

Author: Happy for this to be removed then. Though stating that python 3.x is required would be good I think.

Reviewer: Yes, let's do that. And you can submit a PR to https://github.com/carpentries/workshop-template with this reminder, if it doesn't have one already.

Author: Ok will try and remember to do that.

{: .prereq}

### Windows - [Video tutorial][video-windows]

> 1. Open [https://www.anaconda.com/distribution/][anaconda-windows] with your web browser.

Reviewer: ```suggestion
1. Open <https://www.anaconda.com/products/individual> in your web browser.
```

2. Download the Python 3 installer for Windows.

3. Double-click the executable and install Python 3 using the recommended settings. Make sure that **Register Anaconda as my default Python 3.x** option is checked - it should be in the latest version of Anaconda

### Mac OS X - [Video tutorial][video-mac]

> 1. Visit [https://www.anaconda.com/distribution/][anaconda-mac] with your web browser.

Reviewer: ```suggestion
1. Visit <https://www.anaconda.com/products/individual> in your web browser.
```

2. Download the Python 3 installer for OS X. These instructions assume that you use the graphical installer `.pkg` file.

3. Follow the Python 3 installation instructions. Make sure that the install location is set to "Install only for me" so Anaconda will install its files locally, relative to your home directory. Installing the software for all users tends to create problems in the long run and should be avoided.

### Linux

Note that the following installation steps require you to work from the terminal (shell). 
If you run into any difficulties, please request help before the workshop begins.

> 1.  Open [https://www.anaconda.com/distribution/][anaconda-linux] with your web browser.

Reviewer: ```suggestion
1.  Open <https://www.anaconda.com/products/individual> in your web browser.
```

2.  Download the Python 3 installer for Linux.

3.  Install Python 3 using all of the defaults for installation.

    a.  Open a terminal window.

    b.  Navigate to the folder where you downloaded the installer

    c.  Type

    ~~~

>     $ bash Anaconda3-

Reviewer: ```suggestion
    bash Anaconda3-
```

    ~~~

>     {: .bash}

Reviewer: ```suggestion
    {: .language-bash}
```

>     and press tab.  The name of the file you just downloaded should appear.

Reviewer: ```suggestion
    and press <kbd>Tab</kbd>.  The name of the file you just downloaded should appear.
```

>     d.  Press enter.

Reviewer: ```suggestion
    d.  Press <kbd>Return</kbd>.
```

    e.  Follow the text-only prompts.  When the license agreement appears (a colon

>         will be present at the bottom of the screen) press the space bar until you see the 
>         bottom of the text. Type `yes` and press enter to approve the license. Press 
>         enter again to approve the default location for the files. Type `yes` and 
>         press enter to prepend Anaconda to your `PATH` (this makes the Anaconda 

Reviewer: ```suggestion
        will be present at the bottom of the screen) press <kbd>Spacebar</kbd> until you see the 
        bottom of the text. Type `yes` and press <kbd>Return</kbd> to approve the license. Press 
        <kbd>Return</kbd> again to approve the default location for the files. Type `yes` and 
        press <kbd>Return</kbd> to prepend Anaconda to your `PATH` (this makes the Anaconda 
```

        distribution your user's default Python).

[anaconda]: https://www.anaconda.com/

> [anaconda-mac]: https://www.anaconda.com/download/#macos
> [anaconda-linux]: https://www.anaconda.com/download/#linux
> [anaconda-windows]: https://www.anaconda.com/download/#windows
> [jupyter]: http://jupyter.org/
> [python]: https://python.org

Reviewer: ```suggestion
[jupyter]: https://jupyter.org/
[python]: https://www.python.org/
```

[video-mac]: https://www.youtube.com/watch?v=TcSAln46u9U
[video-windows]: https://www.youtube.com/watch?v=xxQ0mzZ8UvA

## Opening a Conda-enabled Terminal and Verifying the Installation:

### Windows
Click Start, search, or select Anaconda Prompt from the menu. A window should pop up where you can now type commands such as checking your Conda installation with:

~~~
conda --help
~~~
{: .language-bash}

### Mac OSX

Click the Launchpad icon in the Dock, type Terminal in the search field, then click Terminal. 
A window should pop up where you can now type commands such as checking your conda installation with:

~~~
conda --help
~~~
{: .language-bash}

### Linux
This depends a bit on your Linux distribution, but often you will have an Applications listing in which you can select a Terminal icon you can click. A window should pop up where you can now type commands such as checking your conda installation with:

~~~
conda --help
~~~
{: .language-bash}

## Required Python Packages for this Workshop
The following are packages needed for this workshop:

> * [Pandas](http://pandas.pydata.org/)
> * [Jupyter notebook](http://jupyter.org/)
> * [Numpy](http://www.numpy.org/)
> * [Matplotlib](http://matplotlib.org/)

Reviewer: ```suggestion
* [Pandas](https://pandas.pydata.org/)
* [Jupyter notebook](https://jupyter.org/)
* [Numpy](https://numpy.org/)
* [Matplotlib](https://matplotlib.org/)
```

* [Plotnine](https://plotnine.readthedocs.io/en/stable/)

All packages apart from plotnine will have automatically been installed with Anaconda and we can use anaconda as a package manager to install the missing `plotnine` package:
You need to open up a *Terminal*, if you are using Mac OSX, or Linux (see instructions above), or launch an *anaconda-promt*, if you are using Windows. In your terminal window type the following: 

~~~
conda install -y -c conda-forge plotnine
~~~
{: .language-bash}

This will then install the latest version of plotnine into your conda environment. 

## Installing other Packages (Not required)
If you want to install any additional packages, you can do so by opening a terminal window and type:
~~~
conda install -y package_name
~~~
{: .language-bash}

> You may need to install the required packages in this way, if you opted for installing Miniconda, instead of Anaconda. Miniconda is a "light" version of Anaconda. If you install and use Miniconda

Reviewer: ```suggestion
You may need to install the required packages in this way,
if you opted for installing Miniconda, instead of Anaconda.
Miniconda is a lightweight version of Anaconda. If you install and use Miniconda
```
